src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -122,14 +163,16 @@ public void Append(T? value)
                 Buffers.Add(new DataFrameBuffer<T>());
                 NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
             }
-            ReadOnlyDataFrameBuffer<T> lastBuffer = Buffers[Buffers.Count - 1];
+            int bufferIndex = Buffers.Count - 1;
+            ReadOnlyDataFrameBuffer<T> lastBuffer = Buffers[bufferIndex];
             if (lastBuffer.Length == ReadOnlyDataFrameBuffer<T>.MaxCapacity)
             {
                 lastBuffer = new DataFrameBuffer<T>();
                 Buffers.Add(lastBuffer);
                 NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
             }
             DataFrameBuffer<T> mutableLastBuffer = DataFrameBuffer<T>.GetMutableBuffer(lastBuffer);
+            Buffers[bufferIndex] = mutableLastBuffer;
             mutableLastBuffer.Append(value ?? default);
```

Comment:
Do you need to increment `NullCount` if `value` is null?

Reply:
The next line `SetValidityBit` does it. This change is just updating `Buffers[bufferIndex]` with a mutable `DataFrameBuffer`